daemon/audit.c

```diff
@@ -312,6 +328,31 @@ audit_record_from_textfile_new(const char *filename, bool purge)
 	return (AuditRecord *)record;
 }
 
+static bool
+audit_log_check_delimiter(const char *log_file)
+{
+	int fd = -1;
+	char buf[sizeof(AUDIT_DELIMITER)];
```

Comment:
As the AUDIT_DELIMITER define contains a newline,
this should this be buf[sizeof(AUDIT_DELIMITER) + 1] to hold the 0 byte